bangazonapi/views/productlike.py

```diff
@@ -0,0 +1,50 @@
+from rest_framework import serializers
+from rest_framework.viewsets import ViewSet
+from rest_framework.response import Response
+from rest_framework import status
+from bangazonapi.models import Productlike
+from rest_framework.permissions import IsAuthenticatedOrReadOnly
+from rest_framework.parsers import MultiPartParser, FormParser
+from .product import ProductSerializer
+from bangazonapi.models import Product
+
+
+
+class ProductLikeSerializer(serializers.ModelSerializer):
+    """Serializer for ProductLike model"""
+
+    class Meta:
+        model = Productlike 
+        fields = ('id', 'user', 'product')
+
+class ProductLikes(ViewSet):
+    """Request handlers for Product Likes"""
+
+    permission_classes = (IsAuthenticatedOrReadOnly,)
+
+    def create(self, request, product_id=None):
+        """
+        POST operation to like a product.
+        """
+        try:
+            product_like, created = Productlike.objects.get_or_create(product_id=product_id, user=request.user)
+            if created:
+                product_like.save()
+                serializer = ProductLikeSerializer(product_like)
+                return Response(serializer.data)
+            else:
+                return Response({'message': 'Product is already liked.'}, status=status.HTTP_400_BAD_REQUEST)
+        except Productlike.DoesNotExist:
+            return Response({'message': 'Productlike does not exist.'}, status=status.HTTP_404_NOT_FOUND)
```

Comment:
I also noticed that when attempting to POST a `like` to a `product` that does not exist, the server raises the following error:
```
IntegrityError at /products/n/like
FOREIGN KEY constraint failed
```